### ISSUE TITLE
fid selection

### ISSUE THREAD

**Issue by jeanconn** [sot/proseco#43]
Fid selection in the context of proseco requires a little care to maintain speed without sacrificing acquisition margin and science quality.  Sub-tasks (which will break out to issues / PRs)

- [x] Define algorithm, document and present to SSAWG
- [x] Implement basic support code (e.g. get the position of a fid light on ACA CCD).  Use Matlab tools code as the basis for continuity with exiting positions in load products.
- [x] Test support code against flight catalogs.
- [x] Implement algorithm and test.  Current idea is tightly integrated with acq star selection.
  - [x] First part: core code and initial fid catalog selection #49 
  - [x] Integrated fid and acq optimization #59

**Comment by taldcroft** [comment]
I have thought about this quite a lot over the last few days on walks etc and now have a couple of pages of handwritten notes.  It turned out the problem is a little tricky, but I think I have something that will work now.  Basically a heuristic expression of how I would approach catalogs and prioritization from an ACA review perspective in different cases.  Stay tuned...

**Comment by jeanconn** [comment]
Indeed.  I figure the best answer (not commanding fids until after acquisition) is going to take a few years.

**Comment by jeanconn** [comment]
I assume second order effect, but I do you need to use the guide phase dither and the agasc stars to check fids for those kinds of spoilers?

**Comment by jeanconn** [comment]
Also, not sure where fid selection is in master so I don't know how much I should debug the error I'm seeing if I try to pass acqs .
```
In [9]: acqs = proseco.acq.AcqTable.get_acq_catalog(18680)

In [10]: fids = proseco.fid.get_fid_catalog(detector='ACIS-S', acqs=acqs)

...
/fido.real/skare3/lib/python3.6/site-packages/astropy/table/table.py in _init_from_cols(self, cols)
    752         if len(lengths) != 1:
    753             raise ValueError('Inconsistent data column lengths: {0}'
--> 754                              .format(lengths))
    755 
    756         # Set the table masking

ValueError: Inconsistent data column lengths: {0, 1}

```

**Comment by jeanconn** [comment]
It is passing tests, just not my first use case test.  So it looks like it needs some more tests and a bug fix.  My question was less "if it is master should I expect it to work" and more "do you want me to figure this out and fix it or are you working it".

**Comment by jeanconn** [comment]
OK.  I put my bug as a new issue.  I'll also put spoiler detection as an issue.

**Comment by taldcroft** [comment]
Closed by #109, #106, #104, #96, #93, ETC.
